BDArmory/Control/MissileFire.cs

```diff
@@ -1731,15 +1746,55 @@ void OnGUI()
                         case MissileBase.TargetingModes.Inertial:
                             {
                                 MissileBase ml = CurrentMissile;
-                                GUIUtils.DrawTextureOnWorldPos(missile.MissileReferenceTransform.position + (2000 * missile.GetForwardTransform()), BDArmorySetup.Instance.largeGreenCircleTexture, new Vector2(156, 156), 0);
                                 float distanceToTarget = 0;
-                                if (guardTarget)
+
+                                if (!vesselRadarData)
+                                {
+                                    GUIUtils.DrawTextureOnWorldPos(missile.MissileReferenceTransform.position + (2000 * missile.GetForwardTransform()), BDArmorySetup.Instance.largeGreenCircleTexture, new Vector2(156, 156), 0);
+                                    break;
+                                }
+
+                                TargetSignatureData targetData = TargetSignatureData.noTarget;
+                                bool radarTarget = false;
+                                if (_radarsEnabled && vesselRadarData.locked)
+                                {
+                                    targetData = vesselRadarData.lockedTargetData.targetData;
+                                    radarTarget = true;
+                                }
+                                else if (_irstsEnabled)
                                 {
-                                    distanceToTarget = Vector3.Distance(guardTarget.CoM, ml.MissileReferenceTransform.position);
-                                    Vector3 fireSolution = MissileGuidance.GetAirToAirFireSolution(ml, guardTarget.CoM, guardTarget.Velocity());
+                                    targetData = vesselRadarData.activeIRTarget(null, this);
+                                }
+
```

Comment:
No support for Scanned but not Locked targets?
Could be simplified down to something like 
```
if (vesselRadarData)
{
    TargetSignatureData targetData = TargetSignatureData.noTarget;
    if (_radarsEnabled)
    {
        if (vesselRadarData.locked)
            targetData = vesselRadarData.lockedTargetData.targetData;
        else
            targetData = vesselRadarData.detectedRadarTarget(guardTarget != null ? guardTarget : null, this);
    }
    else if (_irstsEnabled)
        targetData = vesselRadarData.activeIRTarget(null, this);

    if (targetData.exists)
    {
        distanceToTarget = Vector3.Distance(targetData.predictedPosition, ml.MissileReferenceTransform.position);
        Vector3 fireSolution = MissileGuidance.GetAirToAirFireSolution(ml, targetData.predictedPosition, targetData.velocity);
        Vector3 fsDirection = (fireSolution - ml.MissileReferenceTransform.position).normalized;
        if (vesselRadarData.locked)
            GUIUtils.DrawTextureOnWorldPos(ml.MissileReferenceTransform.position + (distanceToTarget * fsDirection), BDArmorySetup.Instance.greenDotTexture, new Vector2(6, 6), 0);
        else
            GUIUtils.DrawTextureOnWorldPos(ml.MissileReferenceTransform.position + (distanceToTarget * fsDirection), BDArmorySetup.Instance.greenCircleTexture, new Vector2(36, 36), 5);
        GUIUtils.DrawTextureOnWorldPos(ml.MissileReferenceTransform.position + (distanceToTarget * ml.GetForwardTransform()), BDArmorySetup.Instance.dottedLargeGreenCircle, new Vector2(128, 128), 0);
    }
    else
    {
        GUIUtils.DrawTextureOnWorldPos(missile.MissileReferenceTransform.position + (2000 * missile.GetForwardTransform()), BDArmorySetup.Instance.largeGreenCircleTexture, new Vector2(156, 156), 0);
        break;
    }
}
else
    GUIUtils.DrawTextureOnWorldPos(missile.MissileReferenceTransform.position + (2000 * missile.GetForwardTransform()), BDArmorySetup.Instance.largeGreenCircleTexture, new Vector2(156, 156), 0);
break;
```
if you wanted to differentiate between radar/irst detection